docs/attributes.md

```diff
@@ -0,0 +1,94 @@
+### `#[StateId]`
+
+Link your event to its state(s) with the `StateId` attribute
+
+```php
+class YourEvent extends Event
+{
+    #[StateId(GameState::class)]
+    public int $game_id;
+
+    #[StateID(PlayerState::class)]
+    public int $player_id;
+}
+```
+
+### `#[AppliesToState]`
+
+Another way to link states and events; like [`StateId`](#content-stateid), but using the attributes above the class instead of on each individual id.
```

Comment:
Second argument is the property name, which is implicitly `[state name]_id` unless set (ie. "PlayerState" would assume a `$player_id` property). Third and fourth args are the same as second and third in `StateId`.